datastructures/blockchain.go

```diff
@@ -0,0 +1,79 @@
+package datastructures
+
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+)
+
+// Blockchain is the blockchain datastructure
+type Blockchain struct {
+	Chain  map[int32][]*Block `json:"chain"`
+	Length int32              `json:"length"`
+}
+
+// NewBlockChain initializes a new blockchain struct
+func NewBlockChain() *Blockchain {
+	var b Blockchain
+	b.Chain = make(map[int32][]*Block)
+	return &b
+}
+
+// Get used to get blocks of a specified height
+func (b *Blockchain) Get(height int32) []*Block {
+	if val, ok := b.Chain[height]; ok {
+		return val
+	}
+	return nil
+}
+
+// Insert inserts a block into the blockchain
+func (b *Blockchain) Insert(block *Block) {
+
+	if len(b.Chain[block.Header.Height]) == 0 {
+		var newSlice []*Block = []*Block{block}
```

Comment:
This only occurs when the length of the fork is 0 so it doesn't seem like values would ever be overwritten. The new function also does not seem to initialize the slice for the fork that I'm trying to insert. Maybe because the NewBlockChain function only allocates memory for the map when initialized like this: `&Blockchain{Chain: make(map[int32][]*Block), Length: 0}`